lib/cloudant.js

```diff
@@ -493,14 +499,14 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   };
 
   if (id) {
-    self.updateAttributes(model, id, data, {}, function(err, docs) {
+    self.update(model, {id: id}, data, {}, function(err, docs) {
       if (err && err.statusCode !== 404) return cb(err);
-      else if (err && err.statusCode === 404) {
+      else if (err && err.statusCode === 404 || docs.count === 0) {
```

Comment:
I'd like to see some test coverage that demonstrates that `docs.count` can be zero, otherwise I'd remove this.